src/features/CurrentPostAPI/CurrentPostAPI.ts

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+
+type CurrentPost = {
+  post: Post | null;
+};
+
+const initialState: CurrentPost = {
+  post: null,
+};
+
+const currentPostSlice = createSlice({
+  name: 'currentPost',
+  initialState,
+  reducers: {
+    selectPost: (state, action: PayloadAction<CurrentPost>) => {
```

Comment:
The type for the action payload should be 'PayloadAction<Post>' instead of 'PayloadAction<CurrentPost>'. This is because the payload is expected to be a 'Post' object, not a 'CurrentPost' object.